database/redis/database.go

```diff
@@ -91,23 +92,47 @@ func NewTestDatabaseWithIncorrectConfig(logger moira.Logger) *DbConnector {
 	return NewDatabase(logger, Config{Addrs: []string{"0.0.0.0:0000"}}, testSource)
 }
 
-func (connector *DbConnector) manageSubscriptions(tomb *tomb.Tomb, channel string) (<-chan []byte, error) {
-	c := (*connector.client).Subscribe(connector.context, channel)
-	err := c.Ping(connector.context)
+func (connector *DbConnector) makePubSubConnection(channels []string) (*redis.PubSub, error) {
+	psc := (*connector.client).Subscribe(connector.context, channels...)
+	err := psc.Ping(connector.context)
+	if err != nil {
+		return nil, fmt.Errorf("failed to subscribe to '%s', error: %v", channels, err)
+	}
+	return psc, nil
+}
+
+func (connector *DbConnector) manageSubscriptions(tomb *tomb.Tomb, channels []string) (<-chan []byte, error) {
+	psc, err := connector.makePubSubConnection(channels)
 	if err != nil {
 		return nil, err
 	}
 
 	go func() {
 		<-tomb.Dying()
-		connector.logger.Infof("Calling shutdown, unsubscribe from '%s' redis channels...", channel)
-		c.Unsubscribe(connector.context) //nolint
+		connector.logger.Infof("Calling shutdown, unsubscribe from '%s' redis channels...", channels)
+		psc.Unsubscribe(connector.context) //nolint
 	}()
 
 	dataChan := make(chan []byte, pubSubWorkerChannelSize)
 	go func() {
 		for {
-			raw, _ := c.Receive(connector.context)
+			raw, err := psc.Receive(connector.context)
+			if err != nil {
+				if err == io.ErrUnexpectedEOF {
+					connector.logger.Infof("Receive() returned error: %s", err.Error())
+					continue
+				}
+				connector.logger.Infof("Receive() returned error: %s. Reconnecting...", err.Error())
+				newPsc, err := connector.makePubSubConnection(channels)
+				if err != nil {
+					connector.logger.Errorf("Failed to reconnect to subscription: %v", err)
+					<-time.After(receiveErrorSleepDuration)
+					continue
+				}
+				connector.logger.Info("Reconnected to subscription")
+				psc = newPsc
```

Comment:
Couple questions here: 
- will this variable be picked up by gouroutine with unsubscribing on 112-113 lines?
- shouldn't we unsubscribe first from previous object?

Reply:
> will this variable be picked up by gouroutine with unsubscribing on 112-113 lines?

> shouldn't we unsubscribe first from previous object?

I really don't know :) There is a complete repetition of the logic of the master branch.